RK1

HustonMmmavr opened the pull request:
Сделано апи, проверку код стайла прошли

HustonMmmavr committed:
Added project. TODO 1) fix responses, 2) jpa doesn't work properly 3) Update controllers

Jpa:
1) Repository needs some additional methods
2) JPA drops and creates table every rerun server
3) DataIntegrationViolationException has no infornmation about constraint ?

HustonMmmavr committed:
Error with jpa fixed

HustonMmmavr committed:
Added new messages, TODO: think about messages, repository

1) JPA drops when server rerun
2) Add new messages if it is necessary
3) check signOut, signIn (i think it works)
4) Implement updatePassword, udpateEmail, getRating

HaseProgram committed:
ResponseMaker refactored

HustonMmmavr committed:
Jpa works properly

TODO
1) update user,
2) check signin signout,
3) create controllers
4) add travis,yml
5) Checksyle
6) Heroku deplpy

HustonMmmavr committed:
Added udpdate email

HustonMmmavr committed:
Code cleared, added readme, added swagger

HustonMmmavr committed:
fixed readme

HustonMmmavr committed:
readme fixed again

HustonMmmavr committed:
Fixed messages, added checks

HustonMmmavr committed:
Refactored, checks works

HustonMmmavr committed:
To travis

HustonMmmavr committed:
Fixed pom.xml

HustonMmmavr committed:
Fixed pom.xml

HustonMmmavr committed:
Still fixing pom.xml

HustonMmmavr committed:
Ooops, forgoet change dburl

HustonMmmavr committed:
Trying to fi

HustonMmmavr committed:
Work with db dropped

HustonMmmavr committed:
Unused db files deleted

HustonMmmavr committed:
Fuck

HustonMmmavr committed:
Fixed travis

HustonMmmavr committed:
Travis file changed

HustonMmmavr committed:
Added file to deploy

HustonMmmavr committed:
Fixing code style errors

HustonMmmavr committed:
Checks fixed

HustonMmmavr committed:
Fixed error

HustonMmmavr committed:
Merge pull request #1 from TechnoparkSDD/Wihtout-db

Wihtout db

Badya commented on pom.xml:
```diff
@@ -0,0 +1,126 @@
+    </properties>
+
+    <dependencies>
+        <!--<dependency>-->
```
?

HustonMmmavr replied:
Ну мы сначала подключили бд. Локально все работает, а Тревис не может скомпилировать NoClassDefException вылетает при инициализации jpa. Поэтому пришлось быстро переделывать репрозиторий и сервис на работу со списками.

Badya commented on pom.xml:
```diff
@@ -0,0 +1,126 @@
+            <scope>test</scope>
+        </dependency>
+
+        <!--<dependency>-->
```
?

Badya commented on src/main/java/com/colorit/backend/common/AbstractResponseMaker.java:
```diff
@@ -0,0 +1,99 @@
+ */
+public class AbstractResponseMaker {
+    private static final Map<ViewStatusCode, String> HASH_VIEW_ERROR_AND_FIELD = Map.ofEntries(
+            Map.entry(ViewStatusCode.EMPTY_EMAIL, "email"),
```
каждый элемент енума - это по сути полноценный класс - эти мапы не имеют смысла если в енум добавить поля

HaseProgram replied:
Исправил одну мапу, которая ассоциировала ошибку с именем поля (где ошибка произошла). Насчет остальных - по идее будет нарушение уровней абстракций (UserStatusCode не должен знать с каким http статусом он ассоциирован), можно ли их оставить?

Badya commented on src/main/java/com/colorit/backend/repositories/UserRepositoryList.java:
```diff
@@ -0,0 +1,70 @@
+            //noinspection unused
+            final UserEntity existringUser = getByNickame(userEntity.getNickname());
+            throw new ConstraintNameException();
+        } catch (NoResultException ex) {
```
Использовать иключения для control-flow один из bad practice

Badya commented on src/main/java/com/colorit/backend/services/AbstractServiceResponse.java:
```diff
@@ -0,0 +1,29 @@
+
+import java.util.List;
+
+public class AbstractServiceResponse<T> {
```
тут женерик - дальше абстракция - почему бы не сделать ковариант?

Badya commented on src/main/java/com/colorit/backend/services/UserServiceOnList.java:
```diff
@@ -0,0 +1,83 @@
+import com.colorit.backend.repositories.UserRepositoryList;
+import org.springframework.stereotype.Service;
+
+@Service
```
Аннотация уже не нужна - у вас аннотирован интерфейс

HaseProgram replied:
Удалили аннотацию у интерфейса. Если убрать ее в реализации, спринг не знал, что ему необходимо инициализировать...

Badya replied:
Чтобы заработало надо в конструкторе контроллера использовать интерфейс параметром а не реализацию

Badya commented on src/main/java/com/colorit/backend/views/ResponseView.java:
```diff
@@ -0,0 +1,38 @@
+    private T data;
+
+    public ResponseView() {
+        errors = new HashMap<>();
```
можно инициализацию перенести к объявлению

Badya commented on src/main/java/com/colorit/backend/services/UserServiceOnList.java:
```diff
@@ -0,0 +1,83 @@
+    private final UserRepositoryList userRepository;
+
+    public UserServiceOnList() {
+        userRepository = new UserRepositoryList();
```
можно инициализацию перенести к объявлению

Badya commented on src/main/java/com/colorit/backend/views/ViewStatus.java:
```diff
@@ -0,0 +1,23 @@
+    private final ArrayList<ViewStatusCode> arrayOfErrors;
+
+    ViewStatus() {
+        arrayOfErrors = new ArrayList<>();
```
можно инициализацию перенести к объявлению

Badya commented on src/main/java/com/colorit/backend/views/ViewStatusCode.java:
```diff
@@ -0,0 +1,14 @@
+package com.colorit.backend.views;
+
+@SuppressWarnings("ALL")
+public enum ViewStatusCode {
```
см. коммент про енумы и мапы

HaseProgram replied:
Исправил одну мапу, которая ассоциировала ошибку с именем поля (где ошибка произошла). Насчет остальных - по идее будет нарушение уровней абстракций (UserStatusCode не должен знать с каким http статусом он ассоциирован), можно ли их оставить?

Badya replied:
Почему же - ваш статус вполне может знать в каком виде он используется в HTTP протоколе. Важно понимать что при добавлении нового статуса вам надо не забыть сделать маппинг - а в случае если он уже будет в элементе енума, забыть будет сложней

HustonMmmavr replied:
Ну просто была идея сделать класс ResponseMaker, который знает какие коды сервиса (вьюхи) мапить в соответсвующий httpstaus и сообщение. Просто если все это добавить в соответствующие enum, то особого толку от ResponseMaker уже не будет.

Badya commented on src/test/java/com/colorit/backend/BackendApplicationTests.java:
```diff
@@ -0,0 +1,18 @@
+    @SuppressWarnings("EmptyMethod")
+    @Test
+    public void contextLoads() {
+    }
```
тесты это хорошо - но в данном случае лишнее

HaseProgram replied:
Это было автосгенерировано)

HustonMmmavr committed:
It works but with repository(

HaseProgram committed:
Fixed some issues

HaseProgram committed:
Fixed Service problem

+ some issues

HaseProgram committed:
Fixed checkstyle issues

HaseProgram commented:
Вроде поправили, ждем дальнейших инструкций :)

HustonMmmavr commented:
Ну просто была идея сделать класс ResponseMaker, который знает какие коды сервиса (вьюхи) мапить в соответсвующий httpstaus и сообщение. Просто если все это добавить в соответствующие enum, то особого толку от ResponseMaker уже не будет.

HaseProgram committed:
Fixed service annotation

HaseProgram committed:
[FIX] checkstyle

HustonMmmavr committed:
Update README.md

HustonMmmavr committed:
I dont know is it right

HustonMmmavr committed:
added messages

HustonMmmavr committed:
Fixed checkstyle

HustonMmmavr committed:
Deleted unused import

HustonMmmavr committed:
TODO: additional layer of entity, think about ServiceResponse (4 parts one interface and 3 implementations)

HustonMmmavr committed:
pom fixed, added version hibernate

HustonMmmavr committed:
Merge pull request #2 from TechnoparkSDD/alternative

Alternative

HustonMmmavr commented:
Избавились  от мап и сервис пофиксили. Кстати, с бд разобрался. Действительно, не хватало зависимостей. Я так понимаю сейчас не стоит ее добавлять в этот пул реквест?

HustonMmmavr committed:
Dnb denormalized, todo fix some work in repository

HustonMmmavr committed:
Oooohhh, checkstyle ...

HustonMmmavr committed:
Fixed checkstyle errors

HustonMmmavr committed:
Fixed service response, but i think its not correct

HustonMmmavr committed:
Check style ...

HustonMmmavr committed:
checkstyl again

HustonMmmavr committed:
checkstyle fixed

HustonMmmavr committed:
Fixed, i think its right

HustonMmmavr committed:
Added functions which need to implement, alse need to make class which loads and saves file to FS storage

HustonMmmavr committed:
All views added to associated with them folders; also work with response refactored

HustonMmmavr committed:
Forget about unused imports and cant splits overloaded

HustonMmmavr committed:
Oops

HustonMmmavr committed:
Fixed email validation

HustonMmmavr committed:
Unused imports.. deleted unused dependency

HustonMmmavr committed:
Fixed

Badya commented on src/main/java/com/colorit/backend/common/AbstractResponseMaker.java:
```diff
@@ -0,0 +1,37 @@
+ * @version 1.0
+ */
+@Component
+public class AbstractResponseMaker {
```
Abstract наверное и класс должен быть abstract, тогда и аннотация Сomponent будет не нужна

Badya commented on src/main/java/com/colorit/backend/controllers/AbstractController.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.Locale;
+
+@RestController
+public class AbstractController {
```
Abstract наверное и класс должен быть abstract, тогда и аннотация RestController будет не нужна

Badya commented on src/main/java/com/colorit/backend/entities/AbstractEntity.java:
```diff
@@ -0,0 +1,4 @@
+package com.colorit.backend.entities;
+
+public class AbstractEntity {
```
?

Badya commented on src/main/java/com/colorit/backend/services/AbstractServiceResponse.java:
```diff
@@ -0,0 +1,29 @@
+
+import java.util.List;
+
+public class AbstractServiceResponse<T extends AbstractEntity> {
```
?

Badya commented on src/main/java/com/colorit/backend/repositories/UserRepositoryList.java:
```diff
@@ -0,0 +1,73 @@
+        existringUser.setEmail(email);
+    }
+
+    public static class RepositoryException extends Exception {
```
Зачем checked Exception?

Badya commented on src/main/java/com/colorit/backend/services/UserServiceStatusCode.java:
```diff
@@ -0,0 +1,11 @@
+package com.colorit.backend.services;
+
+public enum UserServiceStatusCode {
```
Это где-то используется?

HustonMmmavr committed:
Deleted useless classes added check by instanceof and renamed classes

HustonMmmavr committed:
Returned work with list

HustonMmmavr committed:
Fixed: unused dependencies deleted,

HustonMmmavr committed:
Checkstyle ...

HustonMmmavr committed:
Fixed, deleted unused files and comments

HustonMmmavr committed:
Merge pull request #3 from TechnoparkSDD/return-db

Return db

Badya commented on src/main/java/com/colorit/backend/common/ResponseMaker.java:
```diff
@@ -0,0 +1,89 @@
+                    return new ResponseEntity<>(new ResponseView<>(
+                            new UserView((UserEntity) userServiceResponse.getData())), httpStatus);
+                } else if (userServiceResponse.getData() instanceof List) {
+                    final List<UserView> userViews = new ArrayList<>();
```
А в каком случае может быть список? почему бы не сделать тогда 2 разные сущности и разнести хи по смыслу?

HustonMmmavr replied:
Ну у нас может вернуться список пользователей (ник, и результат игры для просмотра списка победителей), просто здесь происходит конвертация для каждого объекта листа (объекта из хранилища) в вьюху, которая вернется. Просто если будет сущность UserListView то конвертация UserEntity в UserView будет там. Так сделать?

Badya commented on src/main/java/com/colorit/backend/services/statuses/IStatus.java:
```diff
@@ -0,0 +1,4 @@
+package com.colorit.backend.services.statuses;
+
+public interface IStatus {
```
Пустые интерфейсы это плохой тон, удалить или сделать что-то осмысленное

Badya commented on src/main/java/com/colorit/backend/services/responses/IServiceResponse.java:
```diff
@@ -0,0 +1,15 @@
+import com.colorit.backend.services.statuses.IStatus;
+
+public interface IServiceResponse<T> {
+    void setStatus(IStatus istatus);
```
Если в интерфейсе хочется делать геттеры и сетеры это намек на то что это должен быть абстрактный класс

Badya commented:
Зачем было убирать абстракт - нужно было просто сделать классы абстрактными вы же их нигде не инстанцируете, а так только лишняя объектная путаница

HustonMmmavr committed:
Changed response as abstract

HustonMmmavr committed:
Refactored IStaus and UserServiceStatius; IServiceResponse renamed to AbstractServiceResponse

TODO:
1) Think about entities

HustonMmmavr committed:
fixed

HustonMmmavr committed:
TODO: implement

HustonMmmavr committed:
Added methods, todo implement.

HustonMmmavr committed:
refactor

HustonMmmavr committed:
added funct

HustonMmmavr committed:
TODO: make responses, fileUpload

HustonMmmavr committed:
TODO: improve storag, add methods, finish save image, and other

HustonMmmavr committed:
TODO: implement given functionality (methods, controlers)

HustonMmmavr committed:
getList works, getCount works,

HustonMmmavr committed:
avatar return

HustonMmmavr committed:
TODO: set session, update, fix

HustonMmmavr committed:
fixed work

HustonMmmavr committed:
Added updateView

HustonMmmavr committed:
checkstyle

HustonMmmavr committed:
Update TODO: fix if data is equal existing

HustonMmmavr committed:
Fixed checkstyle

HustonMmmavr committed:
fuck this checkstyle

HustonMmmavr committed:
fixed rating

HustonMmmavr committed:
Fixed ch

HustonMmmavr committed:
Origins returnesds

Badya commented on src/main/java/com/colorit/backend/controllers/AbstractController.java:
```diff
@@ -0,0 +1,32 @@
+
+@RestController
+public abstract class AbstractController {
+    private final @NotNull IUserService userService;
```
Каждому ли контроллеру будет нужен юзер сервис?

HustonMmmavr committed:
fixed initialization userService

HustonMmmavr committed:
Fixed userService initialization

HustonMmmavr committed:
Deleted unused import

HustonMmmavr committed:
Fixed checkstyle

HustonMmmavr committed:
fixed some issues with

HustonMmmavr committed:
Fixed maker, deleted some unused imports,

TODO:
1) AbstractView functions (methods) ?
2) add package for entities
3) delete unused imports

HustonMmmavr committed:
FIxed, added packages

HustonMmmavr committed:
Fixed update nickname

HustonMmmavr committed:
Fixed checkstyle

HustonMmmavr committed:
Merge branch 'refactor'

HustonMmmavr committed:
fixed deploy

HustonMmmavr committed:
fixed unused import

Badya commented on src/main/java/com/colorit/backend/BackendApplication.java:
```diff
@@ -0,0 +1,13 @@
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+//import org.springframework.context.annotation.ComponentScans;
+
+//@ComponentScans(value = )
```
?

Badya commented on src/main/java/com/colorit/backend/controllers/AbstractController.java:
```diff
@@ -0,0 +1,26 @@
+import javax.validation.constraints.NotNull;
+
+@RestController
+@CrossOrigin(origins = {"http://localhost:3000", "https://color-it.herokuapp.com"}, allowCredentials = "true")
```
Наверное стоит сделать параметром

HustonMmmavr committed:
Adedd generator results

Badya commented on src/main/java/com/colorit/backend/views/output/AbstractOutputView.java:
```diff
@@ -0,0 +1,4 @@
+package com.colorit.backend.views.output;
+
+public interface AbstractOutputView {
```
?

Badya commented on src/main/java/com/colorit/backend/repositories/UserRepositoryList.java:
```diff
@@ -0,0 +1,125 @@
+    private final List<UserEntity> db;
+
+    public UserRepositoryList() {
+        db = new ArrayList<>();
```
инициализацию к объявлению

HustonMmmavr committed:
checkstyle

HaseProgram committed:
Fixed save

HustonMmmavr committed:
Ranamed AbstractOutputView IOutputView

HustonMmmavr committed:
Merge branch 'master' of https://github.com/TechnoparkSDD/S.D.D.-02-2018

HustonMmmavr committed:
ResponseView now is not generic

HustonMmmavr committed:
todo: fix

HustonMmmavr committed:
todo;

HustonMmmavr committed:
Fixed

HustonMmmavr committed:
TODO: check it, clear cloudinary, properties

HustonMmmavr committed:
TODO check, clear cloudinary

HustonMmmavr committed:
deleted useless classes, and added crossorigins

HustonMmmavr committed:
Some custom fix

HustonMmmavr committed:
Merge pull request #5 from TechnoparkSDD/global-refactor

Global refactor

HustonMmmavr committed:
Cleared, added work with properties values

HustonMmmavr committed:
Fixed origin

Badya commented on src/main/java/com/colorit/backend/config/Config.java:
```diff
@@ -0,0 +1,43 @@
+public class Config
+{
+    @Autowired
+    Environment mEnv;
```
private?

Badya commented on src/main/java/com/colorit/backend/config/Config.java:
```diff
@@ -0,0 +1,43 @@
+    @Autowired
+    Environment mEnv;
+
+    @Bean(name = "front_url_local")
```
параметры можно прям через @Value делать - @Bean для другого вроде

HustonMmmavr committed:
fixed config

HustonMmmavr committed:
Added file check, crop it. TODO: return db

HustonMmmavr committed:
Fixed checkstyle

HustonMmmavr committed:
Some fix

HaseProgram committed:
Avatar path in response after update

HustonMmmavr committed:
Storage creates by spring

HustonMmmavr committed:
Merge branch 'master' of https://github.com/TechnoparkSDD/S.D.D.-02-2018

HustonMmmavr committed:
Deleted unused imports and checkstyle fixed

HustonMmmavr committed:
Fixed checkstyle again

HustonMmmavr committed:
Added logger

HustonMmmavr committed:
Fixed checkstyle

HustonMmmavr committed:
Fixed checkstyle
